modules/resourceid/fsw/src/cfe_resourceid_api.c

```diff
@@ -121,6 +121,11 @@ CFE_ResourceId_t CFE_ResourceId_FindNext(CFE_ResourceId_t StartId, uint32 TableS
     CFE_ResourceId_t CheckId;
     bool             IsTaken;
 
+    if (CheckFunc == NULL)
+    {
+        return CFE_ES_BAD_ARGUMENT;
+    }
+
```

Comment:
Suggest returning CFE_RESOURCEID_UNDEFINED... otherwise this is a conversion from an error code to a resource id type.